### PR TITLE
Add a button to test configuration of a Payment Gateway #7532

### PR DESCRIPTION
**Pull request by priyankajain18**

### CODE REVIEW

File: tests/test_transaction.py [hunk 1]
```diff
@@ -469,6 +469,20 @@ def test_0240_manual_gateway_capture(self):
...
+            self.setup_defaults()
+
+            gateway, = self.PaymentGateway.create([{
+                'name': 'Test Gateway',
+                'journal': self.cash_journal.id,
+                'provider': 'self',
+                'method': 'manual',
+            }])
+            self.PaymentGateway.test_gateway_configuration([gateway])
+            self.assertEqual(gateway.configured, True)
```
Comment (aroraumang) [inline tests/test_transaction.py]: Test the other case also when there's party required on journal's debit account

File: transaction.py [hunk 2]
```diff
@@ -889,8 +911,8 @@ def default_card_info(self, fields):
         party = Party(Transaction().context.get('active_id'))
 
         res = {
-                'party': party.id,
-                'owner': party.name,
+               'party': party.id,
```
Comment (aroraumang) [inline transaction.py]: Indent

File: view/gateway_form.xml [hunk 3]
```diff
@@ -20,4 +20,7 @@ this repository contains the full copyright notices and license terms. -->
             <field name="users"/>
         </page>
     </notebook>
-</form>
+    <label name="configured"/>
+    <field name="configured"/>
+    <button name="test_gateway_configuration" string="Test Configuration" colspan="2"/>
+    </form>
```
Comment (aroraumang) [inline view/gateway_form.xml]: Indent

File: transaction.py [hunk 4]
```diff
@@ -66,6 +66,28 @@ class PaymentGateway(ModelSQL, ModelView):
...
+                'readonly': ~Bool(Eval('active')),
+            },
+        })
+
+    @classmethod
+    @ModelView.button
+    def test_gateway_configuration(cls, gateways):
+        for gateway in gateways:
+            journal = gateway.journal
+            if journal.debit_account and \
```
Comment (aroraumang) [inline transaction.py]: Replace the if block with this code
```
configured = journal.debit_account and not journal.debit_account.party_required
gateway.configured = configured
gateway.save()
```

**Comment by aroraumang** [comment]
LGTM

File: transaction.py [hunk 5]
```diff
@@ -66,6 +66,28 @@ class PaymentGateway(ModelSQL, ModelView):
     users = fields.Many2Many(
         'payment_gateway.gateway-res.user', 'payment_gateway', 'user', 'Users'
     )
+    configured = fields.Boolean('Configured ?', states=STATES, depends=DEPENDS)
```
Comment (tarunbhardwaj) [inline transaction.py]: IMO this should be readonly alwasy

File: transaction.py [hunk 6]
```diff
@@ -66,6 +66,30 @@ class PaymentGateway(ModelSQL, ModelView):
     users = fields.Many2Many(
         'payment_gateway.gateway-res.user', 'payment_gateway', 'user', 'Users'
     )
+    configured = fields.Boolean(
+        'Configured ?', states=STATES, depends=DEPENDS, readonly=True
```
Comment (tarunbhardwaj) [inline transaction.py]: now no need of states and depends as this record is always readonly

**Pull request merged**